File: src/org/labrad/qubits/resources/MicrowaveBoard.java
package org.labrad.qubits.resources;


public class MicrowaveBoard extends DacBoard {
  private MicrowaveSource uwaveSrc = null;

  public static MicrowaveBoard create(String name) {
    MicrowaveBoard board = new MicrowaveBoard(name);
    return board;
  }

  public MicrowaveBoard(String name) {
    super(name);
  }

  public void setMicrowaveSource(MicrowaveSource uwaves) {
    this.uwaveSrc = uwaves;
  }

  public MicrowaveSource getMicrowaveSource() {
    return uwaveSrc;
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/resources/MicrowaveBoard.scala
package org.labrad.qubits.resources

object MicrowaveBoard {
  def create(name: String): MicrowaveBoard = {
    new MicrowaveBoard(name)
  }
}

class MicrowaveBoard(name: String) extends DacBoard(name) {
  private var uwaveSrc: MicrowaveSource = null

  def setMicrowaveSource(uwaves: MicrowaveSource): Unit = {
    this.uwaveSrc = uwaves
  }

  def getMicrowaveSource(): MicrowaveSource = {
    uwaveSrc
  }
}
